Library/Homebrew/keg.rb

```diff
@@ -102,15 +102,22 @@ def to_s
   MUST_EXIST_DIRECTORIES = (MUST_EXIST_SUBDIRECTORIES + [
     HOMEBREW_CELLAR,
   ].sort.uniq).freeze
+
+  ZSH_DIRECTORIES = (
+  if ENV["HOMEBREW_MULTI_USER"]
+    []
```

Comment:
What happens if an install is attempted and these directories don't exist or aren't writable?

Reply:
@MikeMcQuaid 
- If directories do not exist:  They are created and set with `755` octal permissions  (https://github.com/Homebrew/install/blob/master/install.sh#L402)

- If directories exit but are not writable:  Permissions are updated to `755` 
(https://github.com/Homebrew/install/blob/master/install.sh#L414)

Reply:
Sounds good!